app/cdap/components/hydrator/components/Canvas/index.tsx

```diff
@@ -55,43 +55,50 @@ import { IPipelineComment } from 'components/PipelineCanvasActions/PipelineComme
 import IconSVG from 'components/shared/IconSVG';
 import { copyToClipBoard } from 'services/Clipboard';
 import { INodePosition, ISelectedElements } from './types';
+import { PipelineDiffNode } from './PipelineDiffNode';
 
 interface ICanvasProps {
   angularNodes: any;
   angularConnections: any;
-  isDisabled: boolean;
-  previewMode: boolean;
-  updateNodes: (nodes: any[]) => void;
-  updateConnections: (connections: any[], addStateToHistory?: boolean) => void;
-  onPropertiesClick: (node: any) => void;
-  onMetricsClick: (event: any, node: any, portName?: any) => void;
-  getAngularConnections: () => any;
-  getAngularNodes: () => any;
-  setSelectedNodes: (nodes: any[]) => void;
-  setSelectedConnections: (connections: any[]) => void;
-  onKeyboardCopy: () => void;
-  setPluginActiveForComment: (nodeId: string) => void;
-  getActivePluginForComment: () => string;
-  setPluginComments: (nodeId: string, comments: any) => void;
-  getPluginConfiguration: () => any;
-  getCustomIconSrc: (node: any) => string;
-  shouldShowAlertsPort: (node: any) => boolean;
-  shouldShowErrorsPort: (node: any) => boolean;
-  undoActions: () => void;
-  redoActions: () => void;
-  pipelineComments: IPipelineComment[];
-  setPipelineComments: (val: any) => void;
-  onPreviewData: (event: any, node: any) => void;
-  cleanUpGraph: () => void;
-  pipelineArtifactType: 'cdap-data-pipeline' | 'cdap-data-streams';
-  metricsData: any;
-  metricsDisabled: boolean;
-  redoStates: any[];
-  undoStates: any[];
-  updateNodePositions: (nodePosition: INodePosition) => void;
+  isDisabled?: boolean;
```

Comment:
These types all needed to be optional? were they wrong before?

Reply:
It was done by me to quickly spin up a starter point for Mert to work on. These props are needed for studio page but not required for the diff project

Reply:
This will be reverted after the new canvas component is implemented.